samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -94,7 +95,7 @@
   public LocalApplicationRunner(SamzaApplication app, Config config) {
     this.appDesc = ApplicationDescriptorUtil.getAppDescriptor(app, config);
     isAppModeBatch = new ApplicationConfig(config).getAppMode() == ApplicationConfig.ApplicationMode.BATCH;
-    coordinationUtils = getCoordinationUtils(config);
+    coordinationUtils = getCoordinationUtils(config, getClass().getClassLoader());
```

Comment:
This is actually the plugin classloader also, since it is used to load a `CoordinationUtilsFactory`.
Except for the very initial creation of the classloader, Samza code won't need to deal with framework classloader vs. plugin classloader. That difference is built into the classloader itself, so there is only one classloader that is passed around.

Reply:
Sorry for confusion, my point was that at this point this is currently the framework classloader, but needs to be the plugin classloader. Naming it accordingly will help use the right one later.

I think what you're saying is that this will be addressed in the next PR, and that's a better place to look for "using the appropriate classloader" issues. If so, this is fine as is.

Reply:
This will be changed to the plugin classloader later. Although, we might not have two different classloader variables in here anyways, as we only really need to build and pass around the plugin one.
I think it will indeed be clearer once the follow-up PRs come out.

Reply:
In the case where the new special classloading is not being used, then the classloader that will get used will just come from `getClass().getClassLoader()` instead of using any special classloader instantiation.